src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/CustomSettingsSiteGraphBuilder.cs

```diff
@@ -0,0 +1,80 @@
+using System.Linq;
+using GraphQL.Resolvers;
+using GraphQL.Types;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Options;
+using OrchardCore.Apis.GraphQL;
+using OrchardCore.ContentManagement.GraphQL.Options;
+using OrchardCore.ContentManagement.GraphQL.Queries.Types;
+using OrchardCore.Contents;
+using OrchardCore.CustomSettings.Services;
+
+namespace OrchardCore.ContentManagement.GraphQL.Queries
+{
+    public sealed class CustomSettingsSiteGraphBuilder : ISiteGraphBuilder
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IOptions<GraphQLContentOptions> _contentOptionsAccessor;
+        private readonly CustomSettingsService _customSettings;
+        private readonly IStringLocalizer S;
+
+        public CustomSettingsSiteGraphBuilder(
+            IOptions<GraphQLContentOptions> contentOptionsAccessor,
+            IHttpContextAccessor httpContextAccessor,
+            CustomSettingsService customSettings,
+            IStringLocalizer<DynamicContentTypeBuilder> localizer
+        ) {
+            _httpContextAccessor = httpContextAccessor;
+            _contentOptionsAccessor = contentOptionsAccessor;
+            _customSettings = customSettings;
+
+            S = localizer;
+        }
+
+        public void Build(SiteGraphType siteType)
+        {
+            var serviceProvider = _httpContextAccessor.HttpContext.RequestServices;
+            var contentTypeBuilders = serviceProvider.GetServices<IContentTypeBuilder>().ToList();
+            var settingsTypes = _customSettings.GetAllSettingsTypes();
+
+            foreach (var typeDefinition in settingsTypes)
+            {
+                var typeType = new ContentItemType(_contentOptionsAccessor)
```

Comment:
This is exactly what I was thinking, but then I believe I'd have to recreate a lot of the functionality of the other type builders for building parts and such. Not sure I'll be able to circle back to that right now. I just mocked the settings in my current project since I know they won't change often.